src/main/java/com/online/store/dao/jdbc/impl/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,249 @@
+        Optional<Order> order = getOrderFromOrdersTable(id);
+        return getAndSetProductList(order);
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        String getUserOrdersQuery = "SELECT order_id, user_id FROM orders "
+                + "WHERE user_id = ? AND isDeleted = false";
+        List<Order> usersOrders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(getUserOrdersQuery)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                usersOrders.add(getOrder(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't get orders list with user id: " + userId, e);
+        }
+        for (Order order : usersOrders) {
+            getAndSetProductList(Optional.of(order));
+        }
+        return usersOrders;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        deleteOrderFromOrdersTable(id);
+        deleteProductsFromOrder(getById(id).get());
+        return true;
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> allOrders = getAllOrdersFromOrderTable();
+        for (Order order : allOrders) {
+            getAndSetProductList(Optional.of(order));
+        }
+        return allOrders;
+    }
+
+    private List<Order> getAllOrdersFromOrderTable() {
+        String getAllOrdersQuery = "SELECT * FROM orders WHERE isDeleted = false";
+        List<Order> allOrders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(getAllOrdersQuery)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                allOrders.add(getOrder(resultSet));
+            }
+            return allOrders;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all orders", e);
+        }
+    }
+
+    private boolean deleteOrderFromOrdersTable(Long id) {
+        String deleteOrderQuery = "UPDATE orders SET isDeleted = true WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(deleteOrderQuery)) {
+            statement.setLong(1, id);
+            int numberOfDeletedRows = statement.executeUpdate();
+            return numberOfDeletedRows != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete order with id: " + id, e);
+        }
+    }
+
+    private Optional<Order> getAndSetProductList(Optional<Order> order) {
```

Comment:
```suggestion
    private List<Product> getProductList(Long orderId) {
```
В даний момент ти порушуєш SRP в цьому методі

Reply:
згодна, переробила)